server.py

```diff
@@ -146,6 +143,20 @@ def dashboard():
     contests = Contest.select().order_by(Contest.start_time)
     return bottle.template("dashboard.html", contests=contests)
 
+@app.get("/stats")
+@login_required
+def statistics():
+    sub_history_temp = Submission.select(Contest.code, ContestProblems.question, Submission.time, Submission.is_correct)\
+        .where(Session.token == bottle.request.get_cookie("s_id"))\
+        .join(ContestProblems, on=(Submission.contestProblem == ContestProblems.id)) \
+        .join(Session, on=(Submission.user == Session.user))\
```

Comment:
hmm. this isn't a direct foreign key. perhaps we should change the model? `submission.session.user`?

Reply:
Adding a session as foreign key to submission doesn't look right to me.
Eg.
User logged in. Made a submission with session Id = 12345. Logged out. (That session entry would be deleted from the database on logging out).
So, when we search for that session while calculating stats, it won't be available anymore.